src/navigation/screens/chats/components/ReplyMessageBubble.tsx

```diff
@@ -0,0 +1,286 @@
+import {FontAwesome} from '@expo/vector-icons';
+import {IMessageIPFS} from '@pushprotocol/restapi';
+import React, {useEffect, useState} from 'react';
+import {
+  LayoutChangeEvent,
+  Pressable,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import Globals from 'src/Globals';
+import {usePushApi} from 'src/contexts/PushApiContext';
+import {caip10ToWallet} from 'src/helpers/CAIPHelper';
+
+import {getTrimmedAddress} from '../helpers/chatAddressFormatter';
+import {FileMessageComponent} from './messageTypes';
+import {ImageMessage} from './messageTypes/ImageMessage';
+
+export type ReplyMessageBubbleType = 'replying' | 'replied';
+export type MessengerType = 'SENDER' | 'RECEIVER';
+
+type ReplyMessageBubbleProps = {
+  chatMessage?: IMessageIPFS;
+  componentType: ReplyMessageBubbleType;
+  onCancelReply?: () => void;
+  onLayoutChange?: (nativeEvent: LayoutChangeEvent) => void;
+  reference?: string;
+  chatId?: string;
+  messengerType?: MessengerType;
+  disableCancel?: boolean;
+};
+
+const ReplyMessageBubble = ({
+  chatMessage,
+  componentType,
+  onCancelReply,
+  onLayoutChange,
+  reference,
+  chatId,
+  messengerType,
+  disableCancel,
+}: ReplyMessageBubbleProps) => {
+  const {userPushSDKInstance} = usePushApi();
+  // set and get reply payload
+  const [replyPayloadManager, setReplyPayloadManager] = useState<{
+    payload: IMessageIPFS | null;
+    loaded: boolean;
+    err: string | null;
+  }>({payload: null, loaded: false, err: null});
+
+  // resolve reply payload
+  useEffect(() => {
+    if (!replyPayloadManager.loaded && !replyPayloadManager?.payload) {
+      if (componentType === 'replied') {
+        resolveReplyPayload();
```

Comment:
currently there is a bug in reply preview.

Reply preview loads everytime even when they are loaded earlier and a new message is sent.